File: 1952 three-divisors/solution.cpp
class Solution {
public:
    bool isThree(int n) {
      int root = (int)sqrt(n);
        if (root * root != n) return false; 
        return isPrime(root, 2);            
    }

    bool isPrime(int num, int divisor) {
        if (num < 2) return false;
        if (divisor * divisor > num) return true;
        if (num % divisor == 0) return false;
        return isPrime(num, divisor + 1);
    }
};

File: 1952 three-divisors/solution.rs
impl Solution {
    pub fn is_three(n: i32) -> bool {
       let root: i32 = (n as f64).sqrt() as i32;
        if root*root != n {
            return false
        }
        return Self::isPrime(root, 2)
    }

    pub fn isPrime(num: i32, d:i32) -> bool { 
        if num < 2 {
            return false
        }
        if d*d > num {
            return true
        }
        if num%d == 0 {
            return false
        }
        return Self::isPrime(num, d+1)
    }
}
